### PR TITLE
Issue #817: Support static wildcard import of junit's fail method in RequireFailForTryCatchInJUnitCheck

### PR DESCRIPTION
**Pull request by sebthom**
#817

### CODE REVIEW

**Comment by rnveach** [comment]
@romani Change looks good to me.

File: sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheckTest.java [hunk 1]
```diff
@@ -116,6 +116,14 @@ public void testJunit4And5Mix() throws Exception {
         verify(checkConfig, getPath("InputRequireFailForTryCatchInJunitCheck7.java"), expected);
     }
 
+    @Test
+    public void testStaticWildCardImport() throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(RequireFailForTryCatchInJunitCheck.class);
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
```
Comment (rnveach) [inline sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheckTest.java:123]: Sorry, 1 minor change. Please add an example showing a violation where fail is missing using your wildcard import.

**Comment by rnveach** [comment]
Regression Report: http://rveach.no-ip.org/checkstyle/regression/reports/291/

Violations are removed.

File: sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheckTest.java (hunk 1 above)
Comment (sebthom) [reply sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheckTest.java:123]: Done.

**Comment by sebthom** [comment]
> Regression Report: [http://rveach.no-ip.org/checkstyle/regression/reports/291](http://rveach.no-ip.org/checkstyle/regression/reports/291/)
> 
> Violations are removed.

I don't understand this comment. Also the URL is not reachable.

**Comment by rnveach** [comment]
Try again. Soemtimes my server loses connection.

Regression is to show how your changes impact the check by running it against multiple projects.

**Comment by sebthom** [comment]
> Regression is to show how your changes impact the check by running it against multiple projects.

I checked the regression tests. The changes basically proof that the fix works. You see less false positives now. The missing violations are in classes that use static wildcard import `import static org.junit.Assert.*;`

E.g. http://rveach.no-ip.org/checkstyle/regression/reports/291/hibernate-orm/xref/hibernate-entitymanager/src/test/java/org/hibernate/ejb/test/lock/LockTest.java.html#L107
or http://rveach.no-ip.org/checkstyle/regression/reports/291/RxJava/xref/src/test/java/rx/internal/operators/OperatorDoOnEachTest.java.html#L197

**Pull request merged**